generator.rb

```diff
@@ -0,0 +1,86 @@
+# Portmanteau generator program
+
+# checks letters for if it is a vowel
+def is_vowel?(letter)
+    return letter == "a" || letter == "e" || letter == "i" || letter == "o" || letter == "u"
+end
+
+# validates if word is long enough to continue
+def input_validation
+    # request user input
+    input = gets.chomp
+
+    # word length validation
+    until input.length > 1
+        puts "Please enter more than one character."
+        input = gets.chomp
+    end
+    return input.downcase
+end
+
+# finds last vowel of a word
+def last_vowel(word)
+    i = 0
+    last_vowel = word.length
+  
+    word.each_char do |letter|
+        if is_vowel?(letter)
+            last_vowel = i
+        end
+        i += 1 
+    end
+    # returns word from first letter to letter before last vowel
+    # note for dee ==> I assumed that if the first letter of the first word is a vowel that the whole first word would be used
+    return word[0..last_vowel-1]
+end
+
+# finds first vowel of a word
+def first_vowel(word)
+    i = 0
+    first_vowel = 0
+    
+    word.each_char do |letter|
+        if is_vowel?(letter)
+            first_vowel = i
+            break
+        end
+        i += 1 
+    end
+    # returns word from first vowel to the last letter
+    return word[first_vowel..word.length]
+end
```

Comment:
This method and the other one (`last_vowel`) are fantastic -- I bet they would be even more fantastic if you found a way to get `i` (the index) for free. There's a way to do this in Ruby, actually-- using either `word.split('').each_with_index_ do |letter, i|` or `word.each_char.with_index do |letter, i|`! Then you won't need to set `i = 0` outside of the loop or increment `i += 1` since the Ruby loop will do it automatically.

Reply:
Your code with either of these syntaxes working would look like this:

```ruby
def first_vowel(word)
    first_vowel = 0
    
    word.split('').each_with_index do |letter, i|
        if is_vowel?(letter)
            first_vowel = i
            break
        end
    end
    # returns word from first vowel to the last letter
    return word[first_vowel..word.length]
end
```

or

```ruby
def first_vowel(word)
    first_vowel = 0
    
    word.each_char.with_index do |letter, i|
        if is_vowel?(letter)
            first_vowel = i
            break
        end
    end
    # returns word from first vowel to the last letter
    return word[first_vowel..word.length]
end
```